sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiSinkDefaultValuesIT.java

```diff
@@ -0,0 +1,254 @@
+                      new TableFieldSchema().setName("value").setType("STRING"),
+                      new TableFieldSchema()
+                          .setName("defaultliteral")
+                          .setType("INT64")
+                          .setDefaultValueExpression("42")
+                          .setMode("REQUIRED"),
+                      new TableFieldSchema()
+                          .setName("defaulttime")
+                          .setType("TIMESTAMP")
+                          .setDefaultValueExpression("CURRENT_TIMESTAMP()")
+                          .setMode("REQUIRED")));
+    } else {
+      bqSchema =
+          new TableSchema()
+              .setFields(
+                  ImmutableList.of(
+                      new TableFieldSchema().setName("id").setType("STRING"),
+                      new TableFieldSchema().setName("key2").setType("STRING"),
+                      new TableFieldSchema().setName("value").setType("STRING"),
+                      new TableFieldSchema()
+                          .setName("defaultliteral")
+                          .setType("INT64")
+                          .setDefaultValueExpression("42"),
+                      new TableFieldSchema()
+                          .setName("defaulttime")
+                          .setType("TIMESTAMP")
+                          .setDefaultValueExpression("CURRENT_TIMESTAMP()")));
+    }
+
+    TableSchema sinkSchema = bqSchema;
+    if (!sinkKnowsDefaultFields) {
+      sinkSchema =
+          new TableSchema()
+              .setFields(
+                  bqSchema.getFields().stream()
+                      .filter(tfs -> tfs.getDefaultValueExpression() == null)
+                      .collect(Collectors.toList()));
+    }
+    final TableRow row1 =
+        new TableRow()
+            .set("id", "row1")
+            .set("key2", "bar0")
+            .set("value", "1")
+            .set("defaultliteral", 12);
+    final TableRow row2 = new TableRow().set("id", "row2").set("key2", "bar1").set("value", "1");
+    final TableRow row3 = new TableRow().set("id", "row3").set("key2", "bar2").set("value", "2");
+
+    List<TableRow> tableRows = Lists.newArrayList(row1, row2, row3);
+
+    String tableSpec = createAndGetTablespec(bqSchema);
+    Pipeline p = Pipeline.create();
+
+    BigQueryIO.Write<TableRow> write =
+        BigQueryIO.writeTableRows()
+            .to(tableSpec)
+            .withSchema(sinkSchema)
+            .withNumStorageWriteApiStreams(2)
+            .ignoreUnknownValues()
+            .withTriggeringFrequency(Duration.standardSeconds(1))
+            .withMethod(BigQueryIO.Write.Method.STORAGE_WRITE_API)
+            .withCreateDisposition(BigQueryIO.Write.CreateDisposition.CREATE_NEVER);
+    if (!takeDefault) {
+      write =
+          write.withDefaultMissingValueInterpretation(
+              AppendRowsRequest.MissingValueInterpretation.NULL_VALUE);
```

Comment:
What happens when we specify `MISSING_VALUE_INTERPRETATION_UNSPECIFIED`?

Reply:
added extra validation

Reply:
SGTM we don't have to support that mode for now, it sounds very close to what REQUIRED field mode does anyways